app/src/main/java/org/wikipedia/dataclient/okhttp/CommonHeaderRequestInterceptor.kt

```diff
@@ -4,20 +4,25 @@ import okhttp3.Interceptor
 import okhttp3.Response
 import org.wikipedia.WikipediaApp
 import org.wikipedia.dataclient.RestService
+import org.wikipedia.login.LoginClient
 import java.io.IOException
 
 internal class CommonHeaderRequestInterceptor : Interceptor {
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
         val app = WikipediaApp.instance
+        val url = chain.request().url.toString()
         val builder = chain.request().newBuilder()
                 .header("User-Agent", app.userAgent)
                 .header("X-WMF-UUID", app.appInstallID)
-        if (chain.request().url.encodedPath.contains(RestService.PAGE_HTML_ENDPOINT)) {
```

Comment:
the `url.encodedPath` and `url.host` calls actually involve a lot of string-splitting allocations, so let's just operate on the entire `url.toString()` which is more performant.